### PR TITLE
feat: update node to v20

### LINKED ISSUE #6629: [CW-1973] chore: upgrade node to 18.x
Node 14.xx and 16.xx will reach EOL towards the end of 2023. Switch to node18.x.

Release | Status | Codename | Initial Release | Active LTS Start | Maintenance Start | End-of-life
-- | -- | -- | -- | -- | -- | --
14.x | Maintenance | Fermium | 2020-04-21 | 2020-10-27 | 2021-10-19 | 2023-04-30
16.x | Maintenance | Gallium | 2021-04-20 | 2021-10-26 | 2022-10-18 | 2023-09-11
18.x | LTS | Hydrogen | 2022-04-19 | 2022-10-25 | 2023-10-18 | 2025-04-30
19.x | Current |   | 2022-10-18 | - | 2023-04-01 | 2023-06-01
20.x | Pending |   | 2023-04-18 | 2023-10-24 | 2024-10-22 | 2026-04-30

ref: https://github.com/nodejs/release#release-schedule

<sub>[CW-1973](https://linear.app/chatwoot/issue/CW-1973/chore-upgrade-node-to-18x)</sub>

### PR DESCRIPTION
**Pull request by scmmishra**
Fixes #6629

### CODE REVIEW

**Comment by scmmishra** [comment]
@vn @sojan-official seems like DockerCloud is not working properly, can you folks test this PR to see if I have covered all cases

CleanShot 2023-08-18 at 17 14 15@2x

**Comment by vishnu-narayanan** [comment]
@scmmishra 

```
2023-08-18T07:15:32Z #17 [pre-builder 9/14] RUN yarn install
2023-08-18T07:15:32Z #17 0.431 node: --openssl-legacy-provider is not allowed in NODE_OPTIONS
2023-08-18T07:15:32Z #17 ERROR: executor failed running [/bin/sh -c yarn install]: exit code: 9
2023-08-18T07:15:32Z
2023-08-18T07:15:32Z #4 [internal] load metadata for docker.io/library/ruby:3.2.2-alpine3.16
2023-08-18T07:15:32Z ------
2023-08-18T07:15:32Z > [pre-builder 9/14] RUN yarn install:
2023-08-18T07:15:32Z #17 0.431 node: --openssl-legacy-provider is not allowed in NODE_OPTIONS
2023-08-18T07:15:32Z ------
2023-08-18T07:15:32Z ERROR: failed to solve: executor failed running [/bin/sh -c yarn install]: exit code: 9
2023-08-18T07:15:32Z Build failed using Buildkit (1)
```

**Review by pranavrajs — CHANGES_REQUESTED** [review]
@scmmishra A couple of points here. 

1/ Can't we just go to the latest Node version than 18.17.1?
2/ We need to update the dependencies on [linux installer](https://github.com/chatwoot/chatwoot/blob/develop/deployment/setup_20.04.sh).
3/ How does it work in the development env? Should we just add the NODE_OPTIONS in .bashrc?

**Comment by scmmishra** [comment]
@pranavrajs sorry missed the comments

> 1/ Can't we just go to the latest Node version than 18.17.1?

Sure, I can try this out.

> 2/ We need to update the dependencies on [linux installer](https://github.com/chatwoot/chatwoot/blob/develop/deployment/setup_20.04.sh).

Noted.

> 3/ How does it work in the development env? Should we just add the NODE_OPTIONS in .bashrc?

Doing this will break a lot of NodeJS CLI apps, for instance, `code`, the way to do it is to export the option inline when running the command.

File: deployment/setup_20.04.sh [hunk 1]
```diff
@@ -172,7 +172,7 @@ EOF
 function install_dependencies() {
   apt update && apt upgrade -y
   apt install -y curl
-  curl -sL https://deb.nodesource.com/setup_16.x | bash -
+  curl -sL https://deb.nodesource.com/setup_18.x | bash -
```
[the remark this answers is not in the document]
Comment (scmmishra) [reply deployment/setup_20.04.sh:175]: Yeah, I thought so too. However, `nvmrc` is not a node standard, but of a package manager `nvm`. We can however set the value in `package.json` and access it here

```
"engines": {
    "node": "18.x"
}
```

**Comment by scmmishra** [comment]
Tested the install script on a Ubuntu VM, works fine

CleanShot 2023-08-24 at 17 56 41@2x

**Comment by scmmishra** [comment]
> hey Shivam.. can we also incorporate the changes in the PR as well here? https://github.com/chatwoot/chatwoot/pull/7772/files
> 
> so we can test the cwctl in one go

Sure, I'll merge that branch onto this... The changes there are complete right?

**Comment by sojan-official** [comment]
> https://github.com/chatwoot/chatwoot/pull/7772/files

yep . just need to make sure the node versions match up. also if we are doing the `package.json` parsing for node version

**Comment by scmmishra** [comment]
> > https://github.com/chatwoot/chatwoot/pull/7772/files
> 
> yep . just need to make sure the node versions match up. also if we are doing the `package.json` parsing for node version

Noted, yeah we can fetch the node version from packagejson using `jq` and use

**Comment by scmmishra** [comment]
@sojan-official, there is a problem with detecting the node version from `package.json`. The repo cloning setup happens much after the dependencies are installed. So we won't be able to infer the version from there until we modify the script

**Comment by scmmishra** [comment]
@sojan-official @pranavrajs any idea why Circle CI might be [failing here](https://app.circleci.com/jobs/github/chatwoot/chatwoot/72218) Unable to find the root cause of it, the tests are passing locally

**Comment by pranavrajs** [comment]
- Updated the docker apk install command to use `nodejs-current` instead of `nodejs` to use 20.5 instead of 18. alpine 3.18 `nodejs-current` points to 20.5 and `nodejs` points to 18.17.
- Added an environment variable NODE_VERSION=20.5.0 in Netlify for the storybook builds to pass.
- Added an environment variable NODE_OPTIONS="--openssl-legacy-provider" on CircleCI.

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**